server/routes/api/subscriptions.ts

```diff
@@ -0,0 +1,220 @@
+import invariant from "invariant";
+import Router from "koa-router";
+import subscriptionCreator from "@server/commands/subscriptionCreator";
+import { sequelize } from "@server/database/sequelize";
+import auth from "@server/middlewares/authentication";
+import { Subscription, Event, Document } from "@server/models";
+import { authorize } from "@server/policies";
+import { presentSubscription } from "@server/presenters";
+import { SubscriptionEvent } from "@server/types";
+import { assertIn, assertPresent, assertUuid } from "@server/validation";
+import pagination from "./middlewares/pagination";
+
+const router = new Router();
+
+router.post(
+  "subscriptions.list",
+
+  auth(),
+
+  pagination(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+
+    const { documentId, event } = ctx.body;
+
+    assertPresent(documentId, "documentId is required");
+
+    // Make sure `documentId` is accompanied
+    // with valid subsribable events.
+    assertIn(
+      event,
+      ["documents.update"],
+      `${event} is not a valid subscription event for documents`
+    );
+
+    const subscriptions = await sequelize.transaction(async (transaction) => {
+      const document = await Document.findByPk(documentId, { transaction });
+
+      authorize(user, "listSubscription", document);
+
+      return Subscription.findAll({
+        where: {
+          documentId: document.id,
+          event,
+        },
+        order: [["createdAt", "DESC"]],
+        offset: ctx.state.pagination.offset,
+        limit: ctx.state.pagination.limit,
+        transaction,
+      });
+    });
+
+    ctx.body = {
+      pagination: ctx.state.pagination,
+      data: subscriptions.map(presentSubscription),
+    };
+  }
+);
+
+router.post(
+  "subscriptions.info",
+
+  auth(),
+
+  pagination(),
+
+  async (ctx) => {
+    const { user } = ctx.state;
+
+    const subscriptions = await Subscription.findAll({
```

Comment:
`info` endpoints always return the information for a single record – as such they are not paginated. In this case it should work similar to the subscriptions.list endpoint where a documentId and event are acceptable parameters and it turns the subscription record belonging to the current user (if any).